napalm/nxos_ssh/nxos_ssh.py

```diff
@@ -1008,6 +1008,69 @@ def cli(self, commands):
             cli_output[py23_compat.text_type(command)] = output
         return cli_output
 
+    def get_environment(self):
+        """
+        Get environment facts.
+
+        power and fan are currently not implemented
+        cpu is using 1-minute average
+        """
+
+        environment = {}
+        cpu_cmd = 'show system resources'
+        mem_cmd = 'show system resources'
+        temp_cmd = 'show env temp'
+
+        # cpu
+        output = self._send_command(cpu_cmd)
+        environment.setdefault('cpu', {})
+        environment['cpu'][0] = {}
+        environment['cpu'][0]['%usage'] = 0.0
+        for line in output.splitlines():
+            # Load average:   1 minute: 0.14   5 minutes: 0.18   15 minutes: 0.21
+            match = re.search(r'1 minute: [0-9]+(.[0-9]+)?', line)
+            if match is not None:
+                cpu = match.group(0).split(':')[1]
+                cpu = '{0:.2f}'.format(float(cpu))
+                environment['cpu'][0]['%usage'] = float(cpu)
```

Comment:
A common pattern is to fail if you have not seen the data before, so adding something like the below, right here would help here.

```
            else:
                raise ValueError("Unexpected output from: {}".format(line))
```

Reply:
Good idea, added in eb01e0810ebb2ad222f6350c4ca26ced7f8f2157